sale_order_action_invoice_create_hook/model/sale_order.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.model
+    def _get_invoice_group_key(self, order):
+        return (order.partner_invoice_id.id, order.currency_id.id)
+
+    @api.model
+    def _get_draft_invoices(self, invoices, references):
+        return invoices, references
+
+    @api.multi
+    def action_invoice_create(self, grouped=False, final=False):
```

Comment:
Oh, I now see that you break the inheritance chain. That was already the case in the previous version of course, but there was no discussion or rationale about that on the original PR https://github.com/OCA/sale-workflow/pull/535. As I understand it, hooks need to be monkeypatched in the post_load hook to overwrite the original method from odoo.addons.sale.models.sale.SaleOrder. That way, any inheritance on the method is kept intact.